### PR TITLE
Offload mesh analysis to Job and show activity bar

### LINKED ISSUE #311: Improve user experience for mesh analysis of complex geometries
Originally reported by @rickdalgarno 

Steps to reproduce:
1. Open the attached STL and set the printer to an S5.
2. Scale it down to 25% so that it fits on the build tray.
3. Click on the Smart Slice stage. It takes a long time to process the STL and it appears to the user that something is hung up.

Proposed solutions in order of best to worst:
1. Figure out a way to process the model faster.
2. Messaging to inform user that the STL is loading into SS.

*The STL is the export frame which might be protected under an NDA*

### PR DESCRIPTION
**Pull request by bradymadams**
Fixes #311 

This shows a message and frees up the UI.

### CODE REVIEW

**Comment by bradymadams** [comment]
I need to also test this on project load, but it's mostly ready.

**Review by dougkenik — APPROVED** [review]
Looks good. I know we also release the thread of the UM.Job when we run SmartSliceJobs. Would that help usability here? I don't know if it would cause problems if the user started selecting other options in Smart Slice.

As it is, it seems to work fine, but switching between stages during the geometry analysis is a little slow

**Comment by colman-hartinger** [comment]
There is a bug I found, that causes you not to be able to select after mesh analyses. Getting steps to reproduce. Everything else looks good though

**Comment by bradymadams** [comment]
> Looks good. I know we also release the thread of the UM.Job when we run SmartSliceJobs. Would that help usability here? I don't know if it would cause problems if the user started selecting other options in Smart Slice.
> 
> As it is, it seems to work fine, but switching between stages during the geometry analysis is a little slow

I added a sleep for a second to allow the UI to catch up. I don't really want to release the thread a bunch because it might make the whole process take longer. In my opinion it's better to give the impression we're doing something important so just wait until it's done.

**Comment by colman-hartinger** [comment]
> There is a bug I found, that causes you not to be able to select faces after mesh analyses. Getting steps to reproduce. Everything else looks good though

Just tried 20 or so times to reproduce the bug but couldn't

**Review by colman-hartinger — APPROVED** [review]

**Comment by bradymadams** [comment]
@dougkenik @colman-hartinger I had to make a couple changes to make this all work with loading 3MFs with step data. Take another quick look if you have a chance.

**Review by dougkenik — CHANGES_REQUESTED** [review]
I'm opening an optimized part, attached, and I'm getting this log:

Traceback (most recent call last):
  File "X:\4.6-exe\build\inst\lib\python3.5\site-packages\cura\CuraApplication.py", line 1013, in event
  File "X:\4.6-exe\build\inst\lib\python3.5\site-packages\UM\Qt\QtApplication.py", line 464, in event
  File "X:\4.6-exe\build\inst\lib\python3.5\site-packages\UM\Event.py", line 218, in call
  File "X:\4.6-exe\build\inst\lib\python3.5\site-packages\UM\Signal.py", line 332, in __performEmit
  File "X:\4.6-exe\build\inst\lib\python3.5\site-packages\UM\Workspace\WorkspaceFileHandler.py", line 59, in _readWorkspaceFinished
  File "X:\4.6-exe\build\inst\lib\python3.5\site-packages\UM\Signal.py", line 219, in emit
  File "X:\4.6-exe\build\inst\lib\python3.5\site-packages\UM\Signal.py", line 332, in __performEmit
  File "C:\Users\DougKenik\AppData\Roaming\cura\4.6\plugins\SmartSlicePlugin\SmartSliceExtension.py", line 226, in _getState
    self.proxy.updatePropertiesFromJob(job)
  File "C:\Users\DougKenik\AppData\Roaming\cura\4.6\plugins\SmartSlicePlugin\SmartSliceCloudProxy.py", line 705, in updatePropertiesFromJob
    select_tool.updateFromJob(job)
  File "C:\Users\DougKenik\AppData\Roaming\cura\4.6\plugins\SmartSlicePlugin\select_tool\SmartSliceSelectTool.py", line 91, in updateFromJob
    smart_slice_node.clearFaces()
AttributeError: 'NoneType' object has no attribute 'clearFaces'
[UMS5_Swivel_Mount.zip](https://github.com/tetonsim/is-cura-ui/files/5227177/UMS5_Swivel_Mount.zip)

**Comment by bradymadams** [comment]
@dougkenik my last change should have fixed that.

**Comment by dougkenik** [comment]
It's reading in now, but I get our warning about Modifying Results when I open it on a fresh Cura

**Comment by bradymadams** [comment]
> It's reading in now, but I get our warning about Modifying Results when I open it on a fresh Cura

Can you attach the 3MF  and a screenshot here?

**Comment by dougkenik** [comment]
> > It's reading in now, but I get our warning about Modifying Results when I open it on a fresh Cura
> 
> Can you attach the 3MF and a screenshot here?

Same 3MF as above:

image

**Comment by colman-hartinger** [comment]
> It's reading in now, but I get our warning about Modifying Results when I open it on a fresh Cura

This is happening for any validated or optimized 3mf. Also the "analyzing geometry" message displays on the Prepare stage for the validated and optimized parts, I'm wondering if that could be the cause

File: SmartSlicePlugin/stage/SmartSliceScene.py [hunk 1]
```diff
@@ -432,18 +439,58 @@ class Root(SceneNode):
...
+
+        if exc:
+            Message(
+                title=i18n_catalog.i18n("Unable to analyze geometry"),
+                text=i18n_catalog.i18n("Smart Slice could not analyze the mesh for face selection. It may be ill-formed."),
+                lifetime=0,
+                dismissable=True
+            ).show()
+        elif job.step:
+            self.loadStep(job.step)
```
Comment (colman-hartinger) [inline SmartSlicePlugin/stage/SmartSliceScene.py:491]: getting rid of this line seemed to fix the message popping up on load that Doug was talking about. Not sure what other problems it causes though

**Comment by bradymadams** [comment]
Okay, I think I got that resolved by putting the rest of the project loading into a callback

**Review by dougkenik — APPROVED** [review]
Seems to be working now

**Pull request merged**